### PR TITLE
CSP-566-Added status field in employers,partners,admins,apprentices

### PR DESCRIPTION
**Pull request by YuvrajPatil-DFE**
Added status field in employers, partners admins & apprentices. Updated the testcase for GetApprenticeMember for Apprentice entity. Refactored the namespaces

### CODE REVIEW

File: src/SFA.DAS.AANHub.Application/Apprentices/Queries/GetApprenticeMemberResult.cs [hunk 1]
```diff
@@ -7,6 +7,8 @@ public class GetApprenticeMemberResult
         public Guid MemberId { get; set; }
         public string? Email { get; set; }
         public string? Name { get; set; }
```
Comment (LiamWright) [inline src/SFA.DAS.AANHub.Application/Apprentices/Queries/GetApprenticeMemberResult.cs:9]: I think we're trying to get away from nullable strings where not needed. These properties will always be populated if the Apprentice object is not null

Probably should  updated the string props to:
```suggestion
        public string Name { get; set; } = null!
```

File: src/SFA.DAS.AANHub.Application/Employers/Queries/GetEmployerMemberResult.cs [hunk 2]
```diff
@@ -8,6 +8,8 @@ public class GetEmployerMemberResult
         public string Email { get; set; } = null!;
         public string Name { get; set; } = null!;
         public string Organisation { get; set; } = null!;
+        public string? Status { get; set; }
```
Comment (LiamWright) [inline src/SFA.DAS.AANHub.Application/Employers/Queries/GetEmployerMemberResult.cs:11]: Instead:
```suggestion
        public string Status { get; set; } = null!
```

File: src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberResult.cs [hunk 3]
```diff
@@ -8,6 +8,7 @@ public class GetPartnerMemberResult
         public string Email { get; set; } = null!;
         public string Name { get; set; } = null!;
         public string Organisation { get; set; } = null!;
+        public string? Status { get; set; }
```
Comment (LiamWright) [inline src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberResult.cs:11]: See above comments about nuillability

**Review by LiamWright — COMMENTED** [review]
Comments on nullable properties

File: src/SFA.DAS.AANHub.Application/Employers/Queries/GetEmployerMemberResult.cs (hunk 2 above)
Comment (YuvrajPatil-DFE) [reply src/SFA.DAS.AANHub.Application/Employers/Queries/GetEmployerMemberResult.cs]: Fixed it

File: src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberResult.cs (hunk 3 above)
Comment (YuvrajPatil-DFE) [reply src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberResult.cs]: Fixed it

File: src/SFA.DAS.AANHub.Application/Apprentices/Queries/GetApprenticeMemberResult.cs [hunk 4]
```diff
@@ -5,18 +5,21 @@ namespace SFA.DAS.AANHub.Application.Apprentices.Queries
...
         public static implicit operator GetApprenticeMemberResult?(Apprentice apprentice)
         {
             if (apprentice == null)
                 return null;
 
             return new GetApprenticeMemberResult
             {
-                Email = apprentice.Email,
-                Name = apprentice.Name,
+                Email = apprentice.Email!,
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/Apprentices/Queries/GetApprenticeMemberResult.cs:19]: We need to change entities to have non nullable properties

**Review by DevenKShah — COMMENTED** [review]
Update the entities and columns to be non-nullable for Name and Email in all the tables where applicable. This will then get rid of extra null check in the results conversion.

**Review by ravihejmadi — COMMENTED** [review]
@YuvrajPatil-DFE, looks all good to me. While going through the code I noticed that we are not checking for the membershipstatus for all the entities i.e. apprentice. I can see that Liam has added tests for [employer](https://github.com/SkillsFundingAgency/das-aan-hub-api/blob/b62900aeacf36a8f6307e99b9b340740ef40ea38/src/SFA.DAS.AANHub.Application.UnitTests/Employers/Commands/CreateEmployerMember/CreateEmployerMemberCommandHandlerTests.cs#L23) & [Admin](https://github.com/SkillsFundingAgency/das-aan-hub-api/blob/b62900aeacf36a8f6307e99b9b340740ef40ea38/src/SFA.DAS.AANHub.Application.UnitTests/Admins/Commands/CreateAdminMember/CreateAdminMemberCommandHandlerTests.cs#L24)

**Comment by YuvrajPatil-DFE** [comment]
I have created new PR (https://github.com/SkillsFundingAgency/das-aan-hub-api/pull/56) for CSP-566 as it was getting complicated, issue with merging. The changes were small and easy so replicated these on new branch, hence closing this one.

**Pull request closed without merging**